session/objecthandling.go

```diff
@@ -554,10 +560,66 @@ func UpdateObjectHyperCubeContinuousDataAsync(sessionState *State, actionState *
 			MaxNumberLines: &maxLines,
 		}, false)
 		if err != nil {
-			return errors.Wrapf(err, "failed to get continuous data for object<%s>", obj.ID)
+			return errors.WithStack(err)
+		}
+		return nil
+	}, actionState, true, fmt.Sprintf("failed to get continous data for object<%s>", gob.GenericId))
+}
+
+// UpdateObjectHyperCubeTreeDataAsync send get hypercube tree data request and update saved data
+func UpdateObjectHyperCubeTreeDataAsync(sessionState *State, actionState *action.State, gob *enigma.GenericObject,
+	obj *enigmahandlers.Object, requestDef senseobjdef.GetDataRequests) {
+	sessionState.QueueRequest(func(ctx context.Context) error {
+		sessionState.LogEntry.LogDebugf("Get tree data for object<%s> type<%s>", gob.GenericId, gob.GenericType)
+
+		hypercube := obj.HyperCube()
+		if hypercube == nil {
+			return errors.Errorf("no hybercube found for object<%s> type<%s>", gob.GenericId, gob.GenericType)
+		}
+
+		dimInfo := hypercube.DimensionInfo
+		if len(dimInfo) < 1 {
+			return errors.Errorf("no dimensions found for object<%s> type<%s>", gob.GenericId, gob.GenericType)
+		}
+
+		nodes := make([]*enigma.NxPageTreeNode, 0, len(dimInfo))
+
+		for i, dim := range dimInfo {
+			height := dim.Cardinal
+			if height > requestDef.Height {
+				height = requestDef.Height
+			}
+
+			allValues := true
+			if i == 0 {
+				allValues = false
```

Comment:
allValues := i != 0

Reply:
ah true, a bit cleaner

Reply:
No wanted the code for more generic support for treedata and specifically `allValues` field. This is correct for grid chart which is the only such object right now.

For the select I know what to do now, it's just too much to change right now and want to prioritize other things.

Reply:
Perfect! That's fine; getting this level of support in the gopherciser is very useful not least for the app-evaluator. I agree that the select part is less important.